src/Maker/AbstractDockerMaker.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/*
+ * This file is part of the Symfony MakerBundle package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\MakerBundle\Maker;
+
+use Symfony\Bundle\MakerBundle\ConsoleStyle;
+use Symfony\Bundle\MakerBundle\DependencyBuilder;
+use Symfony\Bundle\MakerBundle\FileManager;
+use Symfony\Bundle\MakerBundle\Generator;
+use Symfony\Bundle\MakerBundle\Util\ComposeFileManipulator;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Yaml\Yaml;
+
+/**
+ * Convenient abstract class for docker makers.
+ *
+ * @author  Jesse Rushlow <jr@rushlow.dev>
+ *
+ * @internal
+ */
+abstract class AbstractDockerMaker extends AbstractMaker
+{
+    protected $fileManager;
+
+    /**
+     * @var ComposeFileManipulator
+     */
+    protected $composeFileManipulator;
+
+    /**
+     * @var string
+     */
+    protected $composeFilePath = '';
+
+    /**
+     * @var bool True if an existing docker-compose.yaml file was found.
+     */
+    protected $existingComposeFile = false;
+
+    /**
+     * @var string Service identifier to be set in docker-compose.yaml.
+     */
+    protected $serviceName = '';
+
+    /**
+     * @var string Version set in docker-compose.yaml for the service. e.g. latest
+     */
+    protected $serviceVersion = '';
+
+    public function __construct(FileManager $fileManager)
+    {
+        $this->fileManager = $fileManager;
+    }
+
+    public function interact(InputInterface $input, ConsoleStyle $io, Command $command): void
+    {
+        $io->section('- Docker Compose Setup-');
+
+        $this->composeFilePath = sprintf('%s/docker-compose.yaml', $this->fileManager->getRootDirectory());
```

Comment:
We can avoid this property (well, at least avoid making it protected - we could make it private) by adding a new protected method in this class to save the Composer file:

```
protected function dumpComposeFile(Generator $generator)
{
    // we do the dumping here
}
```

Reply:
done